### PR TITLE
[Kotlin Migration] CustomStudyDialog

### PR DESCRIPTION
**Pull request by codingtosh**
[Kotlin Migration] CustomStudyDialog
package com.ichi2.anki.dialogs.customstudy

_**Edit by @david-allison-1: for #5225**_

### CODE REVIEW

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
```
com.ichi2.anki.dialogs.CustomStudyDialogTest > learnAheadCardsRegressionTest FAILED
    java.lang.NullPointerException: whatever() must not be null
        at com.ichi2.anki.dialogs.CustomStudyDialogTest.learnAheadCardsRegressionTest(CustomStudyDialogTest.kt:60)

com.ichi2.anki.dialogs.CustomStudyDialogTest > increaseNewCardLimitRegressionTest FAILED
    java.lang.NullPointerException: whatever() must not be null
        at com.ichi2.anki.dialogs.CustomStudyDialogTest.increaseNewCardLimitRegressionTest(CustomStudyDialogTest.kt:101)
```

**Comment by codingtosh** [comment]
Done, PTAL.

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/customstudy/CustomStudyDialog.kt [hunk 1]
```diff
@@ -13,534 +13,509 @@
...
+import com.ichi2.libanki.Deck
+import com.ichi2.libanki.backend.exception.DeckRenameException
+import com.ichi2.utils.HashUtil.HashMapInit
+import com.ichi2.utils.JSONArray
+import com.ichi2.utils.JSONObject
+import com.ichi2.utils.KotlinCleanup
+import timber.log.Timber
+import java.util.*
+
+class CustomStudyDialog(private val collection: Collection?, private val customStudyListener: CustomStudyListener?) : AnalyticsDialogFragment(), TagsDialogListener {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/customstudy/CustomStudyDialog.kt:56]: cleanup: non-null collection

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/customstudy/CustomStudyDialog.kt [hunk 2]
```diff
@@ -13,534 +13,509 @@
...
 
     /**
      * Retrieve the list of ids to put in the context menu list
      * @param dialogId option to specify which tasks are shown in the list
      * @return the ids of which values to show
      */
-    private int[] getListIds(int dialogId) {
-        switch (dialogId) {
-            case CONTEXT_MENU_STANDARD:
+    @KotlinCleanup("Inspect Nullability")
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/customstudy/CustomStudyDialog.kt:343]: cleanup: map dialogID to enum to make exhaustive

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
LGTM, only nits

**Comment by codingtosh** [comment]
@david-allison-1 Done, PTAL.
 
Note: If we make collection non-null in the future, the test will fail with the same error as above: 
```
com.ichi2.anki.dialogs.CustomStudyDialogTest > learnAheadCardsRegressionTest FAILED
    java.lang.NullPointerException: whatever() must not be null
        at com.ichi2.anki.dialogs.CustomStudyDialogTest.learnAheadCardsRegressionTest(CustomStudyDialogTest.kt:60)

com.ichi2.anki.dialogs.CustomStudyDialogTest > increaseNewCardLimitRegressionTest FAILED
    java.lang.NullPointerException: whatever() must not be null
        at com.ichi2.anki.dialogs.CustomStudyDialogTest.increaseNewCardLimitRegressionTest(CustomStudyDialogTest.kt:101)
```

**Review by david-allison-1 — APPROVED** [review]
LGTM

**Review by mikehardy — APPROVED** [review]

**Comment by mikehardy** [comment]
Hi there! **JANUARY** OpenCollective Notice

Just a friendly notice that we try to process OpenCollective payments monthly - it's time for [January 2022 submissions](https://github.com/ankidroid/Anki-Android/pulls?q=is%3Apr+merged%3A2022-01-01..2022-01-31) (I'll follow this with the same process for February in a week)

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

(I only post one comment per person to avoid spamming you, regardless of the number of PRs merged, but this note applies to all PRs merged for in the month of January)

Thanks!

**Pull request merged**